### PR TITLE
patch(cb2-10282): separate RequiredStandard object from SectionIVA

### PR DESCRIPTION
**Pull request by m-mullen**
## Update type definition for new data structure

_One line description_
Additional changes to type def to separate and explicitly define the RequiredStandard object

[CB2-10282](https://dvsa.atlassian.net/browse/CB2-10282)

## Changelog

- Additional changes to type def to separate and explicitly define the RequiredStandard object
- convert Inspection Type to an enum

### CODE REVIEW

File: types/iva/defects/enums/inspectionType.enum.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/* tslint:disable */
+/**
+ * This file was automatically generated by json-schema-to-typescript.
+ * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
+ * and run json-schema-to-typescript to regenerate this file.
+ */
+
+export type InspectionType = "basic" | "normal";
```
Comment (cb-cs) [inline types/iva/defects/enums/inspectionType.enum.ts]: Shouldn't this be an enum now? If you compare it to the EU vehicle category?

**Review by m-coslett — APPROVED** [review]

**Review by cb-cs — APPROVED** [review]

File: types/iva/defects/get/index.d.ts [hunk 2]
```diff
@@ -10,14 +10,15 @@ export type InspectionType = "basic" | "normal";
 export interface SectionIVA {
```
Comment (Daniel-Searle) [inline types/iva/defects/get/index.d.ts:15]: Why are we leaving the inspection type to be a type and not an enum? @m-coslett @cb-cs

**Comment by LGin-BJSS** [comment]
Are we missing an overall type like `DefectGETIVA` to represent the actual return type from the API. i.e the type that contains the Basic and Normal arrays of `SectionIVA[]`?

**Comment by m-mullen** [comment]
Added the DefectGETIVA to the type def but I'm not familiar with the intended structure for IVA and where to put the object def since it cannot all be in index.

File: types/iva/defects/get/index.d.ts (hunk 2 above)
Comment (cb-cs) [reply types/iva/defects/get/index.d.ts:15]: If you look at the conversation in the Romans DVSA chat this is discussed, I believe Michael originally had this as an enum, but when comparing to v3, the enum was only explicitly defined when it was a key/value pair. Anything with just values was defined ine the way it as above.

I guess it comes down to what you decided from your spike, and what the front-end are expecting / agreed to receieve.

File: json-definitions/iva/defects/get/index.json [hunk 3]
```diff
@@ -1,67 +1,36 @@
...
-    "requiredStandards"
+  "required" : [
+    "euVehicleCategory",
+    "basic",
+    "normal"
   ],
   "properties": {
-    "sectionNumber": {
-      "type": ["string"]
+    "euVehicleCategory": {
```
Comment (cb-cs) [inline json-definitions/iva/defects/get/index.json]: There is a slight disparity between the example json provided on the spike, and the interface agreed. Example json references array of string, interface only string.

**Review by Daniel-Searle — APPROVED** [review]

**Review by LGin-BJSS — APPROVED** [review]

**Pull request merged**